### ISSUE TITLE
Update issue Ethos suite 1.5.10 and 1.6.0 on X18 with gyro bootloader 1.4.15 Ethos 1.5.15

### ISSUE THREAD

**Issue by Ze-baloo** [FrSkyRC/ETHOS-Feedback-Community#4918]
Hi all,

i've updated the brand new X20R of a friend to Ethos 1.5.19 under Suite 1.6.0 without any issue.
Once done i've tried to update my X18 with gyro and the update didn't worked as planned.
The update works for the audio files and report a bug while trying to update the firmware and aborts it.
Now, my radio is working but the menus are only in english even if the audio files are updated in french.

In the beginning, i've detected that the internal 128mb memory was full and i cleaned it.
Unfortunately it didn't changed anything.
I've tryed the both repair tools, tried to install Ethos 1.5.19 on the radio without backing up my models and images without any sucess.
I've also tried to install back ethos 1.5.15 but it's not working either.
I've tried on Suite 1.6.0 and 1.5.10 (the one used for my last successfull update on my X18), the update stops also before the end.

I must say that i don't know what to do, i'm starting to think that my X18 may have a hardware issue.

Thanks for any help.

Laurent

**Comment by Nicholas-Luoyi** [comment]
I need more information:
1) What step did you take when you encounter a failure? i.e. flash firmware and system files / flash audio / flash all?
2) What message did you get from Suite when it fails?
3) Would you please turn on the debug mode in Suite and run again. Then open the log directory to post the log file here?

**Comment by Ze-baloo** [comment]
Thanks for your answer. 

1 - As far as i remember, the first time i've tried to flash from 1.5.15 to 1.5.19, i've chosen to write the updated components only.
I'm not sure that i've done it on suite 1.5.10 or 1.6.0.

Later on i've tried to update only the audio files, or the botloader or the firmware alternativy. 

I've tried with my models on and after a factory reset, the result is the same. 

2 - In french i gor the following message : Erreur! L'action a été interrompue.

3 - Ok for the debug mode on suite.
Which one do I run, factory reset or format?
Could you please tell me where is the log file you need?

Laurent

**Comment by Nicholas-Luoyi** [comment]
Point 3)
step1. turn on debug mode
step2. try to update all the components until the error occurs
step3. open logs folder and post the files just generated here. They may have the name starts with main- or debug- or crash-. You need to zip them.
![image](https://github.com/user-attachments/assets/16315fde-53c7-4ee8-9a97-198cbaddd434)

**Comment by Nicholas-Luoyi** [comment]
Also, do you have a sdcard in your X18?

**Comment by Ze-baloo** [comment]
Ok, understood, i'll do it in 6 hours when back home, i'm at the office tjis morning.

And yes, I have a sdcard.
16gb high transfer rate.

**Comment by Ze-baloo** [comment]
As promised, the debug file and some screen captures of before launching the update and another one when it crashed.
I've set ethos suite in english to make it easyer for us in order not to switch languages

Laurent

[debug-20241224.zip](https://github.com/user-attachments/files/18240424/debug-20241224.zip)

**Comment by Nicholas-Luoyi** [comment]
Here I found it stops when copying file to the 128MB flash. Would you please try to do some file operation in the flash, like copying files, deleting files etc. Since I start to worry about the hardware as well. In the meanwhile you could try formatting that flash on PC and do the file operation again.

**Comment by Ze-baloo** [comment]
Ok, thanks. 
If i have time today before leaving for holydays, i'll give a try.
Otherwise i'll need to wait until monday.

Laurent

**Comment by Ze-baloo** [comment]
Nicholas,

We came back a bit earlier today and i jumped on my laptop and my X18 :p

In the 128mb flash i found that the fr.i18n file was corrupted, i was unable to copy it on my latop from the flash.
I went in bootloader mode to have full access to the flash.
I did a repair of the drive with the windows tool and replaced the fr.i18n file by a freshly downloaded one.
Once done i started the update from 1.5.15 to 1.5.19 which ran flawlessly
I think it's ok now

Here is the debug file if you want to confirm.
[debug-20241229.zip](https://github.com/user-attachments/files/18270497/debug-20241229.zip)

Thanks for everything.

Laurent
